client/middleware/asyncMiddleware.js

```diff
@@ -0,0 +1,22 @@
+import * as actionTypes from '../actions/actionTypes';
+import * as actions from '../actions/actions'
+
+const asyncMiddleware = store => next => action => {
+    // Run this middleware for every action that requires async
+    if(action.async) {
+        action.promise
+        .then(value => {
+            store.dispatch(value);
+        });
+    } else {
+        return next(action);
+    }
+}
```

Comment:
1. I'm not convinced this extra layer of abstraction really helps the readability of the code.
2. Why add the extra async and promise values to the actions which have a consistent shape of type and payload. This caused me to think quite a bit about some extra abstractions that really aren't needed. If you really wanted to have this middleware, you could have the same logic with the `type==PROMISE` rather than `async` and the payload rather than an extra data item. Any disadvantage to using payload for the promise?